### ISSUE TITLE
blake2-rfc no longer maintained - change to blake2

### ISSUE THREAD

**Issue by DragonDev1906** [smol-dot/smoldot#1885]
The blake2-rfc dependency hasn't been updated in 3 years, has outdated dependencies and open PRs for those. To me this looks like it is no longer maintained. How about switching to https://crates.io/crates/blake2 (RustCrypto)?

**Comment by tomaka** [comment]
I fundamentally disagree that because a library doesn't get updated means that we should no longer use it.
If anything, a library not getting any update can be a sign of stability/maturity.

**Comment by DragonDev1906** [comment]
That is true. It also leads to increased compile times due to the use of outdated transitive dependencies (even though there is a PR for it) and does not explain why a simple PR updating the readme is over 5 years old and didn't even get a comment.

Besides that there is also a benefit of using a library that has 5x as many users/downloads, especially for security relevant dependencies and the knowledge that if there is an issue it is more likely to get fixed.

In the end it is up to you, I hope you have considered both and actively decided to stay on blake2-rfc.
